internal/controller/workspace/workspace.go

```diff
@@ -151,19 +157,41 @@ func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.E
 		return nil, errors.Wrap(err, errGetPC)
 	}
 
-	tf := c.terraform(dir)
 	switch cr.Spec.ForProvider.Source {
 	case v1alpha1.ModuleSourceRemote:
-		// TODO(negz): Handle errors initialising from a remote module.
-		// Terraform will refuse to init from a remote module if there
-		// are files in the workdir. On the other hand it sometimes
-		// needs these files, e.g. credentials or additional HCL. In the
-		// latter case it will download everything before failing so we
-		// work around the issue by attempting a best-effort init from
-		// a remote module, then adding the files, then later running a
-		// regular local init. The downside to this approach is that we
-		// currently drop errors indicating an invalid remote.
-		_ = tf.Init(ctx, terraform.FromModule(cr.Spec.ForProvider.Module))
+		// NOTE(ytsarev): Retrieve .git-credentials from Spec to /tmp outside of workspace directory
```

Comment:
Are we writing these outside the workspace directory because Terraform won't init if we write them to the workspace directory? Could we just write them to the same dir we use for everything else, or would that be a security concern?

Reply:
@negz we need to write outside of tf dir because go-getter removes the target directory before the git pull. That's probably why `terraform init --from-module` is failing fast if directory non-empty. Similar situation is actually with the `git` itself - it cannot clone repo to the non-empty directory.